FRAuth/FRAuth/Callbacks/IdPCallback.swift

```diff
@@ -0,0 +1,207 @@
+    ///
+    /// - Parameters:
+    ///   - handler: Optional `IdPHandler` instance to perform social login; if not provided, SDK automatically selects the default `IdPHandler` implementation based on `IdPClient.provider` value (`google`, `facebook`, and `apple`; case insensitive)
+    ///   - presentingViewController: Currently presenting `UIViewController` to present modal view for authorization using provider's SDK. For certain providers, `UIViewController` **must be** presented.
+    ///   - completion: Completion callback to notify the result
+    public func signIn(handler: IdPHandler? = nil, presentingViewController: UIViewController? = nil, completion: @escaping SocialLoginCompletionCallback) {
+        
+        //  Keep the handler
+        self.idpHandler = handler
+        
+        //  If IdPHandler is not passed in as parameter, try to get default implementation
+        if self.idpHandler == nil, let defaultHandler = self.getDefaultIdPHandler(provider: self.idpClient.provider) {
+            self.idpHandler = defaultHandler
+        }
+        
+        //  If no handler is passed in as parameter, and no default handler implementation found, throw an exception
+        guard var handler = self.idpHandler else {
+            FRLog.e("Missing handler; given provider (\(self.idpClient.provider)) does not match with any of default IdPHandler implementation, and no IdPHandler is passed into IdPCallback.signIn.")
+            completion(nil, nil, SocialLoginError.missingIdPHandler)
+            return
+        }
+        
+        handler.presentingViewController = presentingViewController
+        handler.signIn(idpClient: self.idpClient) { (token, tokenType, error) in
+            
+            if let token = token, let tokenType = tokenType {
+                self.setToken(token)
+                self.setTokenType(tokenType)
+                FRLog.v("Credentials received - Token Type: \(tokenType) from \(self.idpClient.provider)")
+            }
+            
+            completion(token, tokenType, error)
+        }
+    }
+    
+    
+    //  MARK: - Private
+    
+    /// Retrieves default `IdPHandler` implementation based on the `IdPClient.provider` value
+    /// - Parameter provider: String value of `IdPClient.provider`
+    /// - Returns: `IdPHandler` implementation for the given provider
+    func getDefaultIdPHandler(provider: String) -> IdPHandler? {
+        var idpHandler: IdPHandler?
+        if provider.lowercased().contains("apple") {
+            idpHandler = AppleSignInHandler()
+        }
+        else if provider.lowercased().contains("google") {
+            if let c: NSObject.Type = NSClassFromString("FRGoogleSignIn.GoogleSignInHandler") as? NSObject.Type, let thisHandler = c.init() as? IdPHandler {
+                idpHandler = thisHandler
+            }
+        }
+        else if provider.lowercased().contains("facebook") {
```

Comment:
I would suggest creating a providers enum to get the raw values to compare, instead of using a static string value

Reply:
@george-bafaloukas-forgerock Creating an enumeration value for providers may not make sense in this case as this provider value is plain text input from the server side. One thing that I have not implemented in here is the Callback constants that I had implemented it for other feature you've tested and since that feature branch is still in stash, it is not merged and applied to this feature implementation. 

I would suggest instead of having enum value for the provider value, I prefer to have Callback constants value to compare those three providers.

Let me know what you think.

Reply:
Callback constants is OK with me.